### PR TITLE
[AIRFLOW-4313] Remove the Mesos executor

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. 
  - https://issues.apache.org/jira/browse/AIRFLOW-4313

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Based on the discussion on the dev mailing list: http://mail-archives.apache.org/mod_mbox/airflow-dev/201904.mbox/browser

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by XD-DENG** [comment]
Thanks @kaxil . LGTM. 

May you add a note in the https://github.com/apache/airflow/blob/master/UPDATING.md before I merge? Thanks.

**Comment by kaxil** [comment]
Thanks @XD-DENG , I completely forgot that. Just added now

**Review by XD-DENG — APPROVED** [review]

**Comment by XD-DENG** [comment]
Thanks @kaxil

**Comment by Fokko** [comment]
Thanks @kaxil

File: UPDATING.md [hunk 1]
```diff
@@ -24,6 +24,9 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### Removal of Mesos Executor
+The Mesos Executor is removed from the code base as it was not widely used and not maintained. [Mailing List Discussion on deleting it](https://lists.apache.org/list.html?dev@airflow.apache.org:lte=1M:mesos).
```
Comment (ashb) [inline UPDATING.md:28]: This isn't a permalink btw :)

Comment (kaxil) [reply UPDATING.md:28]: Raised a PR to fix it - https://github.com/apache/airflow/pull/5476 . Thanks :)

**Pull request merged**